data-protocols/dsp/dsp-transfer-process/dsp-transfer-process-transform/src/main/java/org/eclipse/edc/protocol/dsp/transferprocess/transformer/type/to/JsonObjectToTransferCompletionMessageTransformer.java

```diff
@@ -0,0 +1,57 @@
+/*
+ *  Copyright (c) 2023 Fraunhofer Institute for Software and Systems Engineering
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Fraunhofer Institute for Software and Systems Engineering - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.protocol.dsp.transferprocess.transformer.type.to;
+
+import jakarta.json.JsonObject;
+import org.eclipse.edc.connector.transfer.spi.types.protocol.TransferCompletionMessage;
+import org.eclipse.edc.jsonld.transformer.AbstractJsonLdTransformer;
+import org.eclipse.edc.protocol.dsp.spi.types.HttpMessageProtocol;
+import org.eclipse.edc.transform.spi.TransformerContext;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import static java.lang.String.format;
+import static org.eclipse.edc.protocol.dsp.transferprocess.transformer.DspCatalogPropertyAndTypeNames.DSPACE_CALLBACKADDRESS_TYPE;
+import static org.eclipse.edc.protocol.dsp.transferprocess.transformer.DspCatalogPropertyAndTypeNames.DSPACE_PROCESSID_TYPE;
+import static org.eclipse.edc.protocol.dsp.transferprocess.transformer.DspCatalogPropertyAndTypeNames.DSPACE_TRANSFER_COMPLETION_TYPE;
+
+public class JsonObjectToTransferCompletionMessageTransformer extends AbstractJsonLdTransformer<JsonObject, TransferCompletionMessage> {
+
+    public JsonObjectToTransferCompletionMessageTransformer() {
+        super(JsonObject.class, TransferCompletionMessage.class);
+    }
+
+    @Override
+    public @Nullable TransferCompletionMessage transform(@NotNull JsonObject jsonObject, @NotNull TransformerContext context) {
+        var type = nodeType(jsonObject, context);
+
+
+        if (DSPACE_TRANSFER_COMPLETION_TYPE.equals(type)) {
+
+            var transferCompletionMessageBuilder = TransferCompletionMessage.Builder.newInstance();
+
+            transferCompletionMessageBuilder
+                    .protocol(HttpMessageProtocol.DATASPACE_PROTOCOL_HTTP);
+
+            transformString(jsonObject.get(DSPACE_PROCESSID_TYPE), transferCompletionMessageBuilder::processId, context);
+            transformString(jsonObject.get(DSPACE_CALLBACKADDRESS_TYPE), transferCompletionMessageBuilder::callbackAddress, context);
+
+            return transferCompletionMessageBuilder.build();
+        } else {
+            context.reportProblem(format("Cannot transform type %s to TransferRequestMessage", type));
```

Comment:
Why could it not transform? Please add information that will be useful in diagnosing problems

Reply:
Removed